cmd/thanos/tools_stream_metric.go

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/oklog/run"
+	"github.com/opentracing/opentracing-go"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/thanos-io/thanos/pkg/extkingpin"
+	"github.com/thanos-io/thanos/pkg/store/storepb"
+
+	"github.com/prometheus/prometheus/tsdb/chunkenc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type rawMetricConfig struct {
+	storeAddr            string
+	metric               string
+	hoursAgo             int
+	hours                int
+	skipChunks           bool
+	ignoreWarnings       bool
+	equalLabelMatcher    string
+	notEqualLabelMatcher string
+}
+
+func registerFlags(cmd extkingpin.FlagClause) *rawMetricConfig {
+	conf := &rawMetricConfig{}
+	cmd.Flag("store", "Thanos Store API gRPC endpoint").Default("localhost:10901").StringVar(&conf.storeAddr)
+	cmd.Flag("metric", "The metric name to stream time series for this metric").Default("up").StringVar(&conf.metric)
+	cmd.Flag("hours_ago", "Stream the metric from this number of hours ago").Default("2").IntVar(&conf.hoursAgo)
+	cmd.Flag("hours", "Stream data samples for this number of hours").Default("2").IntVar(&conf.hours)
```

Comment:
Let's just use minT maxT terminology here, would be more consistent with StoreAPI.

Reply:
done

Reply:
Might be missing something here, but the flags still seem to be named hours and hours_ago?